core/src/avm2/activation.rs

```diff
@@ -1550,6 +1570,26 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         Ok(FrameControl::Continue)
     }
 
+    fn op_newcatch(
+        &mut self,
+        method: Gc<'gc, BytecodeMethod<'gc>>,
+        index: Index<Exception>,
+    ) -> Result<FrameControl<'gc>, Error> {
+        if let Some(body) = method.body() {
+            let ex = &body.exceptions[index.0 as usize];
+            let vname = ex.variable_name;
+            let qname = QName::from_abc_multiname(
+                method.translation_unit(),
+                vname,
+                self.context.gc_context,
+            )?;
+            let so = ScriptObject::catch_scope(self.context.gc_context, &qname);
```

Comment:
Instead of adding a function to `ScriptObjectData`, I think you can directly do:
```rs
let so = ScriptObject::custom_object(self.context.gc_context, None, None);
let vt = VTable::newcatch(self.context.gc_context, &qname); // see note below
so.set_vtable(self.context.gc_context, vtable);
so.install_instance_slots(&mut self);
```

and, a general note: As long as the class design looks like it looks currently, we probably should still create a real ClassObject for this scope. The reason is that while the bytecode can't use the class directly, it can kinda-sorta observe things like its name using `getQualifiedClassName`.
Either way, we should also probably load them AOT instead of creating them on the fly... though admittedly it will make things like error type name resolution harder.
Anyway, both of these can be done later; maybe just add a comment about that?

Reply:
I prefer to keep this function. It keeps the low-level implementation of the catch scope with ScriptObject, which I think is more relevant. (Not a strong preference though).

Comment added.